packages/clients/src/aptos/aptos.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2023-2024 Shinami Corp.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
+
+const NODE_REST_URL = "https://api.shinami.com/aptos/node/v1";
+const NODE_INDEXER_URL = "https://api.shinami.com/aptos/graphql/v1";
+
+/**
+ * A private function which infers the network enum from a Shinami access key
+ * @param accessKey
+ * @returns a Network enum type or undefined
+ */
+function inferNetworkFromKey(accessKey: string): Network | undefined {
+  if (accessKey.startsWith("aptos_devnet")) {
+    return Network.DEVNET;
+  } else if (accessKey.startsWith("aptos_testnet")) {
+    return Network.TESTNET;
+  } else if (accessKey.startsWith("aptos_mainnet")) {
```

Comment:
For the sake of completeness (and when working with local env), let's also include `aptos_local`.